System.cpp

```diff
@@ -17,9 +17,30 @@ Node* System::getChild(Dir* dir) const {
 }
 
 Node* System::stringToNode(std::string name) {
-    for (auto &n : currentFolder->getChildren()) {
-        if (n->getName() == name) {
-            return n;
+    if (name.find("/") != std::string::npos) {
+        Dir* current = currentFolder;
+        std::string work;
+
+        try
+        {
+            work = goToFolder(name);
+        }
+        catch(NoDirectoryExc e) 
+        {
+            std::cerr << e.getWhat() << '\n';
+        }
+        for (auto &n : currentFolder->getChildren()) {
+            if (n->getName() == work) {
+                currentFolder = current;
+                return n;
+            }
+        }
```

Comment:
ezt nem teljesen ertem, hogy mit csinal, illetve ha a try nem sikerult akkor is cisnalja?!